src/platform/K32W/BLEManagerImpl.cpp

```diff
@@ -37,7 +58,1415 @@ namespace Internal {
+        snprintf(mDeviceName, kMaxDeviceNameLength, "%s%04u", CHIP_DEVICE_CONFIG_BLE_DEVICE_NAME_PREFIX, discriminator);
+    }
+
+    /**************** Prepare advertising data *******************************************/
+    adv.cNumAdStructures = BLEKW_ADV_MAX_NO;
+
+    chipErr = ConfigurationMgr().GetBLEDeviceIdentificationInfo(mDeviceIdInfo);
+    SuccessOrExit(chipErr);
+    mDeviceIdInfoLength = sizeof(mDeviceIdInfo);
+
+    if ((mDeviceIdInfoLength + CHIP_ADV_SHORT_UUID_LEN + 1) > BLEKW_MAX_ADV_DATA_LEN)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    adv_data[0].length = 0x02;
+    adv_data[0].adType = gAdFlags_c;
+    adv_data[0].aData  = (uint8_t *) (&chipAdvDataFlags);
+
+    adv_data[1].length = static_cast<uint8_t>(mDeviceIdInfoLength + CHIP_ADV_SHORT_UUID_LEN + 1);
+    adv_data[1].adType = gAdServiceData16bit_c;
+    memcpy(advPayload, ShortUUID_CHIPoBLEService, CHIP_ADV_SHORT_UUID_LEN);
+    memcpy(&advPayload[CHIP_ADV_SHORT_UUID_LEN], (void *) &mDeviceIdInfo, mDeviceIdInfoLength);
+    adv_data[1].aData = advPayload;
+
+    adv.aAdStructures = adv_data;
+    /**************** Prepare scan response data *******************************************/
+    scanRsp.cNumAdStructures = BLEKW_SCAN_RSP_MAX_NO;
+
+    scan_rsp_data[0].length = static_cast<uint8_t>(strlen(mDeviceName) + 1);
+    scan_rsp_data[0].adType = gAdCompleteLocalName_c;
+    scan_rsp_data[0].aData  = (uint8_t *) mDeviceName;
+
+    scan_rsp_data[1].length = sizeof(chipOverBleService) + 1;
+    scan_rsp_data[1].adType = gAdComplete16bitServiceList_c;
+    chipOverBleService[0]   = ShortUUID_CHIPoBLEService[0];
+    chipOverBleService[1]   = ShortUUID_CHIPoBLEService[1];
+    scan_rsp_data[1].aData  = (uint8_t *) chipOverBleService;
+
+    scanRsp.aAdStructures = scan_rsp_data;
+
+    /**************** Prepare advertising parameters *************************************/
+    advInterval =
+        ((NumConnections() == 0 && !ConfigurationMgr().IsPairedToAccount()) || GetFlag(mFlags, kFlag_FastAdvertisingEnabled))
+        ? (CHIP_DEVICE_CONFIG_BLE_FAST_ADVERTISING_INTERVAL * 3)
+        : CHIP_DEVICE_CONFIG_BLE_SLOW_ADVERTISING_INTERVAL;
+
+    adv_params.advertisingType = gAdvConnectableUndirected_c;
+    adv_params.ownAddressType  = gBleAddrTypePublic_c;
+    adv_params.peerAddressType = gBleAddrTypePublic_c;
+    memset(adv_params.peerAddress, 0, gcBleDeviceAddressSize_c);
+    adv_params.channelMap   = (gapAdvertisingChannelMapFlags_t)(gAdvChanMapFlag37_c | gAdvChanMapFlag38_c | gAdvChanMapFlag39_c);
+    adv_params.filterPolicy = gProcessAll_c;
+    adv_params.minInterval = adv_params.maxInterval = advInterval;
+
+    err = blekw_start_advertising(&adv_params, &adv, &scanRsp);
+    if (err == BLE_OK)
+    {
+        ChipLogProgress(DeviceLayer, "Started Advertising.");
+    }
+    else
+    {
+        ChipLogProgress(DeviceLayer, "Advertising error!");
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+exit:
```

Comment:
is the exit label used anywhere?

This function seems to return right away and not use the exit label pattern. I would suggest using only one approach (either return anywhere, or have an exit lable, not both).